### DISCUSSION TITLE
Disable count query when using Specification and Pageable | Blaze Persistence EntityView

### DISCUSSION THREAD

**Discussion by TheKerbycz** [Blazebit/blaze-persistence#1850] [Q&A]
Hello, is it possible to disable count query when using Specification and Pageable? I tried setting property `PAGINATION_DISABLE_COUNT_QUERY` to `true` on the `EntityViewConfiguration`, but unfortunately, by looking into the code, is seems like the query is always executed with count query enabled. [see line 679 in AbstractEntityViewAwareRepository](https://github.com/Blazebit/blaze-persistence/blob/b440469b31a8a297fc14eb88dae6feee1a6ec55e/integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/repository/AbstractEntityViewAwareRepository.java#L620C30-L620C30).

One solution could be to implement new class ConfigurablePageable extending the Pageable interaface and let user explicitly specify the count query behavior .

Thank you for any suggestions.

**Comment by beikov** [comment]
Hi, you can use a `EntityViewSettingProcessor` in your Spring Data repository method. Also see the last section of the [features chapter](https://persistence.blazebit.com/documentation/1.6/entity-view/manual/en_US/index.html#spring-data-features).

```java
@Transactional(readOnly = true)
public interface SimpleCatViewRepository extends Repository<Cat, Long> {
    List<SimpleCatView> findAll(EntityViewSettingProcessor<SimpleCatView> processor);
}
```

```java
simpleCatViewRepository.findAll(setting -> setting.withProperty(PAGINATION_DISABLE_COUNT_QUERY, "true"));
```
